drake/solvers/NloptSolver.cpp

```
@@ -0,0 +1,291 @@
+  // with >1 output.
+  for (auto& c : prog.getGenericConstraints()) {
+    WrappedConstraint wrapped = { c.getConstraint().get(),
+                                  &c.getVariableList() };
+    wrapped_list.push_back(wrapped);
+    std::vector<double> tol(c.getConstraint()->getNumConstraints(), 1e-8);
+    opt.add_inequality_mconstraint(
+        EvaluateVectorConstraint, &wrapped_list.back(), tol);
+  }
+
+  // sam.creasey: The initial implementation of this code used
+  // add_equality_mconstraint to handle constraints with multiple
+  // outputs as a vector constraint.  This did not seem to work.  The
+  // version below breaks out the problem into multiple constraints.
+  std::list<LinearEqualityConstraint> equalities;
+  for (auto& c : prog.getLinearEqualityConstraints()) {
+    const size_t num_constraints = c.getConstraint()->getNumConstraints();
+    const auto& A = c.getConstraint()->getMatrix();
+    const auto& b = c.getConstraint()->getLowerBound();
+    for (size_t i = 0; i < num_constraints; i++) {
+      equalities.push_back(LinearEqualityConstraint(A.row(i), b.row(i)));
+      WrappedConstraint wrapped = { &equalities.back(),
+                                    &c.getVariableList() };
+      wrapped_list.push_back(wrapped);
+      opt.add_equality_constraint(
+          EvaluateScalarConstraint, &wrapped_list.back(), 1e-8);
+    }
+  }
+
+  // TODO sam.creasey Missing test coverage for generic constraints
+  // with >1 output.
+  for (auto& c : prog.getLinearConstraints()) {
+    WrappedConstraint wrapped = { c.getConstraint().get(),
+                                  &c.getVariableList() };
+    wrapped_list.push_back(wrapped);
+    std::vector<double> tol(c.getConstraint()->getNumConstraints(), 1e-8);
+    opt.add_inequality_mconstraint(
+        EvaluateVectorConstraint, &wrapped_list.back(), tol);
+  }
+
+  opt.set_xtol_rel(1e-6);
+
+  double minf = 0;
+  opt.optimize(x, minf);
```

It seems to reliably throw in the event of failure.  I was going to skip checking the result unless I had some particular reason to handle it.  The alternative would be to eat the exception at this level and return false.